src/clients/c++/perf_client.cc

```diff
@@ -1374,15 +1376,23 @@ InferenceProfiler::SummarizeLatency(
 
   summary.client_avg_latency_ns = tol_latency_ns / latencies.size();
 
-  if (report_percentile_) {
+  if (extra_percentile_) {
     // Round to nearest integer index by + 0.5
     size_t index = (percentile_ / 100.0) * (latencies.size() - 1) + 0.5;
-    summary.client_percentile_latency_ns = latencies[index];
-    summary.reporting_latency_ns = summary.client_percentile_latency_ns;
+    summary.client_requested_percentile_latency_ns = latencies[index];
+    summary.reporting_latency_ns = summary.client_requested_percentile_latency_ns;
   } else {
     summary.reporting_latency_ns = summary.client_avg_latency_ns;
   }
 
+  // retrieve other interesting percentile
```

Comment:
I can't find the right line, but where you say "Reporting xxx latency" at the beginning to indicate what you are using the stabilize the latency, "reporting" isn't the correct work anymore since you are reporting many latencies. Perhaps change that to "Stabilizing using average latency", or "Stabilizing using p90 latency".